terraform_compliance/steps/steps.py

```diff
@@ -201,15 +202,21 @@ def its_key_is_value(_step_obj, key, value):
         elif isinstance(object_key, list) and value in object_key:
             found_list.append(obj)
 
-        elif isinstance(object_key, dict) and (value in object_key.keys()):
-            found_list.append(obj)
+        elif isinstance(object_key, dict):
+            if value in object_key.keys():
+                if dict_value is None or (object_key[value] == dict_value):
```

Comment:
knit picking: You can have all these in one line.

Reply:
Yep, but it'll do for now?